src/devtoolsPanel.ts

```diff
@@ -179,7 +181,50 @@ export class DevToolsPanel {
             sourceMaps: `${this.config.sourceMaps}`,
         });
 
-        // TODO: Parse message and open the requested file
+        // Parse message and open the requested file
+        const { column, line, url, omitFocus } =
+            JSON.parse(message) as { column: number, line: number, url: string, omitFocus: boolean };
+
+        // If we don't want to focus on the doc and doing so would cause a tab switch ignore it.
+        // This is because just starting to edit a style in the Elements tool with call openInEditor
+        // but if we switch vs code tab the edit will be cancelled.
+        if (omitFocus && this.panel.viewColumn === vscode.ViewColumn.One) {
+            return;
+        }
+
+        // Convert the devtools url into a local one
+        let sourcePath = url;
+        if (this.config.sourceMaps) {
+            sourcePath = applyPathMapping(sourcePath, this.config.sourceMapPathOverrides);
+        }
+
+        // Convert the local url to a workspace path
+        const transformer = new debugCore.UrlPathTransformer();
+        transformer.launch({ pathMapping: this.config.pathMapping });
+        const localSource = { path: sourcePath };
+        await transformer.fixSource(localSource);
+
+        sourcePath = localSource.path || sourcePath;
+
+        // Convert the workspace path into a VS Code url
+        let uri: vscode.Uri;
+        try {
+            uri = vscode.Uri.file(sourcePath);
+        } catch {
+            uri = vscode.Uri.parse(sourcePath, true);
+        }
+
+        // Finally open the document if it exists
+        if (uri) {
+            const doc = await vscode.workspace.openTextDocument(uri);
+            vscode.window.showTextDocument(
+                doc,
+                {
+                    preserveFocus: true,
+                    selection: new vscode.Range(line, column, line, column),
+                    viewColumn: vscode.ViewColumn.One,
+                });
+        }

```

Comment:
Should we maybe have some sort of error message if this doesn't work? I feel like users might be confused if nothing happens.

Reply:
We should, it's just not easy to do that right now without adding some logging logic. Filed #85 to track this work